daemon/emer-daemon.c

```diff
@@ -1123,6 +1142,379 @@ set_max_bytes_buffered (EmerDaemon *self,
   priv->max_bytes_buffered = max_bytes_buffered;
 }
 
+static void schedule_next_midnight_tick (EmerDaemon *self);
+static void schedule_next_month_tick (EmerDaemon *self);
+
+static GVariant *
+aggregate_timer_to_variant (AggregateTimerData *timer_data,
+                            guint32             previous_counter)
+{
+  gboolean complete = timer_data->end_monotonic_us != -1;
+  guint32 counter;
+  gint64 end_time;
+
+  counter = previous_counter;
+
+  end_time = complete ? timer_data->end_monotonic_us : g_get_monotonic_time ();
+  counter += (guint32) end_time - timer_data->start_monotonic_us;
+
+  return g_variant_new ("(vvvuubmv)",
```

Comment:
If there is documentation about what the fields of this structure are, I can't see it. What is the definition of 'complete' for example?

Reply:
As you wrote on https://phabricator.endlessm.com/T32203:

> At midnight², e-e-r-d adds any running timers to the corresponding tally [...] and then adds what I'll call **"complete"** aggregate events based on the data in the tally

Which I interpreted as aggregate events that had their timer stopped.

Reply:
Thinking about it a bit more: at midnight (resp end-of-month) we iterate across all running timers, add the time elapsed so far to the on-disk tally, and reset their start time. Only after that do we submit to the server. So I think we can say that any data that is in the on-disk tally at the point where we iterate across it all is "complete" in this sense.